Gifski/TrimmingAVPlayerViewController.swift

```diff
@@ -52,12 +59,33 @@ final class TrimmingAVPlayerView: AVPlayerView {
 	private var timeRangeObserver: NSKeyValueObservation?
 	private var trimmingObserver: NSKeyValueObservation?
 
+	/// The minimum duration the trimmer can be set to.
+	var minimumTrimDuration = 0.1
+
 	fileprivate func observeTrimmedTimeRange(_ updateClosure: @escaping (ClosedRange<Double>) -> Void) {
+		var skipNextUpdate = false
+
 		// Observing `.duration` seems buggy on macOS 10.14.
 		// Once we change minimum target to 10.15,
 		// observe `\.duration` instead of `\.forwardPlaybackEndTime`.
 		timeRangeObserver = player?.currentItem?.observe(\.forwardPlaybackEndTime, options: .new) { item, _ in
-			guard let playbackRange = item.playbackRange else {
+			guard
+				let fullRange = item.range,
+				let playbackRange = item.playbackRange
+			else {
+				return
+			}
+
+			/// Prevent infinite recursion.
+			guard !skipNextUpdate else {
+				skipNextUpdate = false
+				updateClosure(playbackRange.minimumRangeDiff(of: self.minimumTrimDuration, in: fullRange))
+				return
+			}
+
+			guard playbackRange.interval > self.minimumTrimDuration else {
```

Comment:
Try commenting out `skipNextUpdate = true`, run the app, and the drag the right trim handle all the way to the left. It will go into an infinite recursion and crash. This logic prevents that. There might be a better way to handle it, but I couldn't think of it yet.